internal/checker/checker.go

```diff
@@ -10826,39 +10967,15 @@ func (c *Checker) isPropertyAccessible(node *ast.Node, isSuper bool, isWrite boo
 }
 
 func (c *Checker) containerSeemsToBeEmptyDomElement(containingType *Type) bool {
-	if c.compilerOptions.Lib == nil || slices.Contains(c.compilerOptions.Lib, "lib.dom.d.ts") {
+	return !slices.Contains(c.compilerOptions.Lib, "dom") && everyContainedType(containingType, hasCommonDomTypeName) && c.isEmptyObjectType(containingType)
+}
+
+func hasCommonDomTypeName(t *Type) bool {
+	if t.symbol == nil {
 		return false
 	}
-	return everyContainedType(containingType, func(t *Type) bool {
-		if t.symbol == nil {
-			return false
-		}
-
-		name := t.symbol.Name
-
-		switch name {
-		case "EventTarget", "Node", "Element":
-			return true
-		}
-
-		name, ok := strings.CutPrefix(name, "HTML")
-		if !ok {
-			return false
-		}
-
-		name, ok = strings.CutSuffix(name, "Element")
-		if !ok {
-			return false
-		}
-
-		for _, r := range name {
-			if !stringutil.IsASCIILetter(r) {
-				return false
-			}
-		}
-
-		return true
-	}) && c.isEmptyObjectType(containingType)
+	name := t.symbol.Name
+	return name == "EventTarget" || name == "Node" || name == "Element" || strings.HasPrefix(name, "HTML") && strings.HasSuffix(name, "Element")
```

Comment:
The code that was submitted here was a more faithful port, but I guess it's probably not a problem?

Reply:
Yeah, little point in wasting code for the ASCII-only character check.